.gitignore

```diff
@@ -3,3 +3,6 @@
 /dist
 /yt.egg-info
 __pycache__
+__config__.py
+*.so
+*.c
```

Comment:
This is definitly a problem, because in developer mode several `*.c` files are created (like `yt/geometry/grid_container.c`) and should not become part of the git repo but other `*.c` files are part of the original yt repo (like `yt/analysis_modules/halo_finding/fof/fof_main.c`).